pypsa/optimization/constraints.py

```diff
@@ -98,24 +100,32 @@ def define_operational_constraints_for_extendables(
     attr : str
         name of the attribute, e.g. 'p'
     """
-    lhs_lower: DataArray | tuple
-    lhs_upper: DataArray | tuple
-
     ext_i = n.get_extendable_i(c)
+    com_i = n.get_committable_i(c)
+    mod_i = n.df(c).query(f"({nominal_attrs[c]}_mod>0)").index
 
-    if ext_i.empty:
+    inter_i = ext_i.difference(n.get_committable_i(c)).rename(ext_i.name)
+
+    if inter_i.empty:
+        inter_i = ext_i.intersection(com_i).difference(mod_i).rename(ext_i.name)
+    else:
+        inter_i = inter_i.union(
+            ext_i.intersection(com_i).difference(mod_i).rename(ext_i.name)
+        )
```

Comment:
Probably a good idea to explain this.
We may find a compact way to do this, maybe with a smart query search; but let's keep this for later

Reply:
Yes, I agree, and initially I repeated the same function twice to avoid things like this. I agree that repeating the same function twice is not a good option, but I will find a way to handle this.